docs/study-def-tricks.md

```diff
@@ -0,0 +1,318 @@
+This section introduces some tricks for working more efficiently with the study definition API, by helping you to avoid copy-and-pasting chunks of code unnecessarily. 
```

Comment:
I think we need to add a few words to warn that these are all python examples and will probably be hard to adapt for someone without a bit of python experience?

I think we could add something similar in this part in the main Study Def page, and link to this "hacks" page:

> It is written in the Python programming language, using an OpenSAFELY-specific format which is intended to be easily written, read, and reviewed by anyone with some epidemiological knowledge.

Something like "when you find you are copy and pasting a lot, you may find it easier to write loops in python code, which does require a little more python skill - examples here"...?

Reply:
We should also invite people who come up with their own hacks to share them with us! It's a great way of learning common patterns that we should consider integrating into the framework.